src/main/java/no/unit/nva/handlers/AddUserHandler.java

```diff
@@ -0,0 +1,54 @@
+package no.unit.nva.handlers;
+
+import static nva.commons.utils.attempt.Try.attempt;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import java.util.Optional;
+import no.unit.nva.database.DatabaseService;
+import no.unit.nva.database.DatabaseServiceImpl;
+import no.unit.nva.exceptions.DataHandlingError;
+import no.unit.nva.model.UserDto;
+import nva.commons.exceptions.ApiGatewayException;
+import nva.commons.handlers.RequestInfo;
+import nva.commons.utils.Environment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AddUserHandler extends HandlerWithEventualConsistency<UserDto, UserDto> {
+
+    private static final Logger logger = LoggerFactory.getLogger(AddUserHandler.class);
+    public static final String UNEXPECTED_FAILRE_MESSAGE = "Failure while trying to get existing user";
+    public static final String SYNC_ERROR_MESSAGE = "Error while trying to retrieve saved user:";
+    private final DatabaseService databaseService;
+
+    /**
+     * Default constructor.
+     */
+    public AddUserHandler() {
+        this(new Environment(), new DatabaseServiceImpl());
+    }
+
+    public AddUserHandler(Environment environment,
+                          DatabaseService databaseService) {
+        super(UserDto.class, environment, logger);
+        this.databaseService = databaseService;
+    }
+
+    @Override
+    protected UserDto processInput(UserDto input, RequestInfo requestInfo, Context context) throws ApiGatewayException {
+        databaseService.addUser(input);
+
+        return getEventuallyConsistent(() -> getUser(input))
```

Comment:
Spending time waiting in a Lambda is not good idea (costs money), do we really need to wait for eventual consistency?

Reply:
Maybe it doesn't matter much for our scale.